crawler/dockercontainer.py

```diff
@@ -72,13 +87,27 @@ def poll_docker_containers(timeout, user_list=None, host_namespace=''):
     containers.
     :return: a DockerContainer object (just the first container created).
     """
-
-    """
-    TODO: implement this function. For now, just emulate a failed poll:
-    no containers were created before timeout seconds.
-    """
     if timeout > 0:
-        time.sleep(timeout)
+        eventQ = multiprocessing.JoinableQueue()
+        eventProc = multiprocessing.Process(
+            name='docker-monitor',
+            target=start_polling, args=(eventQ, ))
```

Comment:
To your first point, Yes, thats what I had described above as "ideal" solution. We want a single instance of monitor which shares a queue with dockercontainer polling interface.